tests/test_test_generator.py

```diff
@@ -10,106 +11,228 @@
-        NetworkTestCase(
-            ClusterHost(allowed_namespace, forbidden_labels),
-            ClusterHost(allowed_namespace, {}),
-            "*",
-            False,
+        pytest.param(
+            [k8s.client.V1Namespace(metadata=k8s.client.V1ObjectMeta(name="default"))],
+            [
+                k8s.client.V1NetworkPolicy(
+                    metadata=k8s.client.V1ObjectMeta(
+                        name="allow-labelled-pods", namespace="default"
+                    ),
+                    spec=k8s.client.V1NetworkPolicySpec(
+                        pod_selector=k8s.client.V1LabelSelector(match_labels={}),
+                        ingress=[
+                            k8s.client.V1NetworkPolicyIngressRule(
+                                _from=[
+                                    k8s.client.V1NetworkPolicyPeer(
+                                        pod_selector=k8s.client.V1LabelSelector(
+                                            match_labels={
+                                                "test.io/test-123_XYZ": "test_456-123.ABC"
+                                            }
+                                        )
+                                    )
+                                ]
+                            )
+                        ],
+                    ),
+                )
+            ],
+            [
+                NetworkTestCase(
+                    ClusterHost(
+                        "default", {"test.io/test-123_XYZ": "test_456-123.ABC"}
+                    ),
+                    ClusterHost("default", {}),
+                    "*",
+                    True,
+                ),
+                NetworkTestCase(
+                    ClusterHost(
+                        "default",
+                        {
+                            INVERTED_ATTRIBUTE_PREFIX
+                            + "test.io/test-123_XYZ": "test_456-123.ABC"
+                        },
+                    ),
+                    ClusterHost("default", {}),
+                    "*",
+                    False,
+                ),
+                NetworkTestCase(
+                    ClusterHost(
+                        INVERTED_ATTRIBUTE_PREFIX + "default",
+                        {"test.io/test-123_XYZ": "test_456-123.ABC"},
+                    ),
+                    ClusterHost("default", {}),
+                    "*",
+                    False,
+                ),
+                NetworkTestCase(
+                    ClusterHost(
+                        INVERTED_ATTRIBUTE_PREFIX + "default",
+                        {
+                            INVERTED_ATTRIBUTE_PREFIX
+                            + "test.io/test-123_XYZ": "test_456-123.ABC"
+                        },
+                    ),
+                    ClusterHost("default", {}),
+                    "*",
+                    False,
+                ),
+            ],
+            id="Allow labelled pods to namespace",
```

Comment:
Should we use a "real" sentence here to make it easier to understand what tested?

Reply:
What kind of sentence would you expect? So far these ids only describe the network policy. Should I just enhance that description?

Reply:
`Allow labelled pods to communicate with Pods in the default namespace` since this is what we are testing here but I'm also fine with the current id.

Reply:
I changed all to a proper sentence